TypeCobol.LanguageServer/TypeCobolCustomLanguageServer/TypeCobolCustomLanguageServer.cs

```diff
@@ -189,22 +203,46 @@ protected virtual void OnDidReceiveSignatureHelpContext(SignatureHelpContextPara
         /// <param name="bForced">Force the server to send the program OutlineNodes</param>
         protected virtual void OnDidReceiveRefreshOutline(string uri, bool bForced)
         {
-            var context = GetDocumentContextFromStringUri(uri, Workspace.SyntaxTreeRefreshLevel.NoRefresh);
-
-            if (context != null && context.FileCompiler != null)
+            if (this.UseOutlineRefresh)
             {
-                var refreshOutlineParams = context.LanguageServer.UpdateOutline(context.FileCompiler.CompilationResultsForProgram.ProgramClassDocumentSnapshot, bForced);
-                if (refreshOutlineParams != null)
+                var context = GetDocumentContextFromStringUri(uri, Workspace.SyntaxTreeRefreshLevel.NoRefresh);
+
+                if (context != null && context.FileCompiler != null)
                 {
-                    SendOutlineData(refreshOutlineParams);
+                    var refreshOutlineParams = context.LanguageServer.UpdateOutline(context.FileCompiler.CompilationResultsForProgram.ProgramClassDocumentSnapshot, bForced);
+                    if (refreshOutlineParams != null)
+                    {
+                        SendOutlineData(refreshOutlineParams);
+                    }
                 }
             }
+        }
 
+        /// <summary>
+        /// Update Cfg/Dfa information to the client.
+        /// </summary>
+        /// <param name="uri">Uri of the document to update Cfg/Dfa Informations</param>
+        private void UpdateCfgDfaInformation(string uri)
+        {
+            if (this.UseCfgDfaDataRefresh)
+            {
+                var context = GetDocumentContextFromStringUri(uri, Workspace.SyntaxTreeRefreshLevel.NoRefresh);
+                if (context != null && context.FileCompiler != null)
+                {
+                    var cfgDfaParams = context.LanguageServer.UpdateCfgDfaInformation(context, !(InRobotLsrTestMode || InLsrClientMode));
```

Comment:
It'd better to have a dedicated option to de/activate the dotFile.
Because a client without a mean to visualize the dot file don't care about it.

```suggestion
                    var cfgDfaParams = context.LanguageServer.UpdateCfgDfaInformation(context, writeCfgDfaDotFile);
```

Reply:
This hack is only for LSR Test mode,  and these are options set thru LSR execution mode, it is not a Server option. An this is used for CFGViewer to have a file ready to read. This is a caller's choice.

Reply:
I agree with @smedilol, the choice here for the client is how the result should be returned (as a separate file or inside JSON response). It is not really related to the use of the robot because we can imagine scenarios where we are monitoring messages between client and server and the client would still use file mode.

Reply:
So what you say is that there is an option in the client to select file or content, and the option shall be applied to LSR mode also, So this is another issue.

Reply:
No I say we forget about LSR mode because it is of no concern here :
- we need a new option `-cfg` to let the client choose whether it wants to have the CFG/DFA notifications
  - this option let also the client choose the nature of results, for example `-cfg AsFile` for temporary dot file and `-cfg AsContent` for inline JSON
- unit tests are not compatible with `-cfg AsFile` because the file path is unpredictable, they shall use `-cfg AsContent`

And no need to change anything about `-r` option and its related properties.

Reply:
Ok, let's create a new issue for this new option, and tests that come with it.
LSR client execution mode will use -cfg AsContent.

Reply:
It's not a new issue, it's a remark about a change in this PR, so it must be done in this PR.

Reply:
Ok, it's not a new issue, but it's a new option.
For TypeCobol.CLI Option or for the Language Server Only?,
What is the default value  AsFile or AsContent ?
Why it should not be possible to pass an output file for the AsFile options? because even tests can create their own output file and compare the result as it is done in other tests that generates a file.

Reply:
This is for Language Server only it is an extension of the new `-cfg` option that you just introduced in this PR.

I'm not sure about the default value but you can also make it mandatory if it's easier.

Passing the output file path cannot be done (easily) as an option because there are multiple documents opened at once therefore we need multiple paths and some kind of mapping between source files and dot files. It can be done with a request though. Instead of saying 'I support CFG/DFA notifications', the client would request CFG/DFA data directly and could specify in the request the file in which it expects the result.

Reply:
Ok in this case no output file as option.